Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Fleet/Registration.hs

```diff
@@ -46,19 +65,88 @@ data FleetOwnerLoginReq = FleetOwnerLoginReq
   }
   deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
 
+data FleetOwnerRegisterReq = FleetOwnerRegisterReq
+  { firstName :: Text,
+    lastName :: Text,
+    mobileNumber :: Text,
+    mobileCountryCode :: Text,
+    merchantId :: Text,
+    city :: City.City,
+    fleetType :: Maybe FOI.FleetType,
+    panNumber :: Maybe Text,
+    gstNumber :: Maybe Text,
+    panImageId1 :: Maybe Text,
+    panImageId2 :: Maybe Text
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+newtype FleetOwnerRegisterRes = FleetOwnerRegisterRes
+  { personId :: Text
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
 newtype FleetOwnerVerifyRes = FleetOwnerVerifyRes
   { authToken :: Text
   }
   deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
 
+fleetOwnerRegister :: FleetOwnerRegisterReq -> Flow FleetOwnerRegisterRes
+fleetOwnerRegister req = do
+  let merchantId = ShortId req.merchantId
+  mobileNumberHash <- getDbHash req.mobileNumber
+  deploymentVersion <- asks (.version)
+  merchant <-
+    QMerchant.findByShortId merchantId
+      >>= fromMaybeM (MerchantNotFound merchantId.getShortId)
+  merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just req.city)
+  let personAuth = buildFleetOwnerAuthReq merchant.id req
+  person <-
+    QP.findByMobileNumberAndMerchantAndRole req.mobileCountryCode mobileNumberHash merchant.id DP.FLEET_OWNER
+      >>= maybe (createFleetOwnerDetails personAuth merchant.id merchantOpCityId True deploymentVersion.getDeploymentVersion) return
+  createFleetOwnerInfo person.id merchant.id req.fleetType req.gstNumber
```

Comment:
Will it not createFleetOwnerInfo everytime register API is being called? Shouldn't be handled like you handle for `createFleetOwnerDetails` or move this also inside `createFleetOwnerDetails`?

Reply:
yes , fixed.